### PR TITLE
Make it easier to store a model entirely in ES

### PR DESCRIPTION
**Pull request by ebarendt**
Simply `include ElasticRecord::LoadFromSource` in the model.

No need for models to set the flag manually. Enables `Model.first`,
`find` and friends for easy lookup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,14 +224,13 @@ end
 
 ```
 
-Use `elastic_index.load_from_source = true` to configure an index without ActiveRecord.
+Include `ElasticIndex::LoadFromSource` to configure an index without ActiveRecord.
```
Comment (malept) [inline README.md]: Mention that finder methods on the model will get delegated to `ElasticRelation`.

File: lib/elastic_record/load_from_source.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  module LoadFromSource
+    extend ActiveSupport::Concern
+
+    included do
+      elastic_index.load_from_source = true
+    end
+
+    module ClassMethods
+      delegate :find, :find_by, :find_each, :find_in_batches, :find_ids_in_batches, :first, to: :elastic_relation
+    end
```
Comment (matthuhiggins) [inline lib/elastic_record/load_from_source.rb]: I would like to see https://github.com/data-axle/content_system/blob/master/app/models/concerns/index_scope_methods.rb dealt with too.

File: README.md [hunk 3]
```diff
@@ -224,14 +224,13 @@ end
...
 
 ```ruby
 class Product
   include ActiveModel::Model
   include ElasticRecord::Record
-
-  self.elastic_index.load_from_source = true
+  include ElasticRecord::LoadFromSource
```
Comment (matthuhiggins) [inline README.md]: 👎 Use a DSL.

Comment (ebarendt) [reply README.md]: Are opposed to the module or how it's included here?

I could add a `loaded_from_source` class method that calls `include`. Then we'd have:

```ruby
class Product
  include ActiveModel::Model
  include ElasticRecord::Record
  loaded_from_source
end
```

File: lib/elastic_record/load_from_source.rb (hunk 2 above)
Comment (ebarendt) [reply lib/elastic_record/load_from_source.rb]: The whole module or the delegations?

File: README.md (hunk 3 above)
Comment (bullfight) [reply README.md]: `elastic_index.load_from_source!`

I like this. It is very similar to the current `elastic_index.load_from_source = true` and the other index configurations we provide via `elastic_index`.

File: lib/elastic_record/load_from_source.rb (hunk 2 above)
Comment (ebarendt) [reply lib/elastic_record/load_from_source.rb]: I gave it a try, but ran into problems with things that are ActiveRecord models but also a PercolatorModel. Pagination got screwed up.

**Comment by ebarendt** [comment]
I think this is good now. I spent a bunch of time on trying to move the code out of that other module, but I need to move on.

File: README.md (hunk 3 above)
Comment (matthuhiggins) [reply README.md]: Only opposed to adding the module as the way to get the behavior. I am suggesting that `elastic_index.load_from_source!` includes the module.

File: lib/elastic_record/index.rb [hunk 4]
```diff
@@ -65,6 +65,11 @@ def enable!
       @disabled = false
     end
 
+    def load_from_source!
+      @load_from_source = true
```
Comment (matthuhiggins) [inline lib/elastic_record/index.rb]: Use `self.load_from_source`, in case the implementation does more than set a var?

**Comment by ebarendt** [comment]
Thoughts on moving the `delegate :query, :filter, :aggregate` to here though?

**Comment by matthuhiggins** [comment]
1. Some kind of way to specify which methods are delegated to, or groupings of methods:

* :finders
* :querying
* :calculations
* :batches

```ruby
elastic_index.active_record_overrides :finders, :batches
```

2. Shorten `Place.elastic_search` to `Place.es`. Require uses of `.es` to access methods such as `count`. Of course, `Place.filter(..).count` would continue to work.

**Pull request merged**